### PR TITLE
cyrus_sasl: 2.1.27 -> 2.1.28

### LINKED ISSUE #157803: sssd + AD/LDAP stopped working after update to 21.11 or unstable due to misconfigured SASL
### Describe the bug
sssd + AD/LDAP stopped working after update to 21.11 or unstable due to misconfigured SASL

### Steps To Reproduce
1. Update from 21.05 to 21.11
2. Attempt to log in

### Expected behavior
Log in works

### Additional context
Errors in logs are:
```
No worthy mechs found
SASL: No worthy mechs found
ldap_sasl_interactive_bind_s failed (-6)[Unknown authentication method]
Unsable to establish connection [number]: Authentication failed
```

### Notify maintainers
@e-user @illustris 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: "x86_64-linux"
- host os: "Linux 5.10.88, NixOS, 21.05.4719.0fd9ee1aa36 (Okapi)'
- multi-user?: `yes`
- sandbox: `yes`
- version `nix-env (Nix) 2.3.16`
- channels(root): `"nixos-21.05.4719.0fd9ee1aa36"`
- nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by illustris**
###### Motivation for this change

- Bump version
- fetch source from GitHub as suggested in the [documentation](https://www.cyrusimap.org/sasl/sasl/developer/installation.html)
- update cyrus-sasl-ac-try-run-fix.patch
- remove unnecessary patches

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by illustris** [comment]
Upstream appears to be getting ready to put out a new stable release. It might be better to wait another week for that rather than apply patches to an older version.

File: pkgs/development/libraries/cyrus-sasl/default.nix [hunk 1]
```diff
@@ -34,6 +34,11 @@ stdenv.mkDerivation rec {
       url = "https://sources.debian.org/data/main/c/cyrus-sasl2/2.1.27+dfsg-1+deb10u1/debian/patches/0021-CVE-2019-19906.patch";
       sha256 = "1n4c5wg7l9j8rlbvx8i605j5d39xmj5wm618k8acxl4fmglcmfls";
     })
+    (fetchpatch { # https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1003355
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cyrus-sasl/default.nix:37]:
```suggestion
    # https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1003355
    (fetchpatch {
```

**Comment by SuperSandro2000** [comment]
> Upstream appears to be getting ready to put out a new stable release. It might be better to wait another week for that rather than apply patches to an older version.

Sure we can wait for this but you don't know if the release is usable right at start.

Anyhow, this needs to target staging.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by illustris** [comment]
closing in favor of https://github.com/NixOS/nixpkgs/pull/161417

**Pull request closed without merging**